src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioServiceImpl.java

```diff
@@ -123,46 +129,157 @@ public BinaryData encode(Integer mediaId) {
+            if (audioToVersion.isEmpty()) {
+                List<MediaEntity> audioFileToProcess = new ArrayList<MediaEntity>();
+                MediaEntity newEntity = mapper.mapToMedia(addAudioMetadataRequest);
+
+                audioFileToProcess.add(newEntity);
+                audioToVersion = Optional.of(audioFileToProcess);
+
+                log.info("No duplicates found. Uploading new file");
             } else {
-                externalLocation = dataManagementApi.saveBlobDataToInboundContainer(digestInputStream);
-                checksum = fileContentChecksum.calculate(digestInputStream);
+                log.info("Identified media ids that require duplicate audio");
+            }
+
+            // version the file upload to the database
+            versionUpload(audioToVersion.get(), addAudioMetadataRequest,
+                           externalLocation, checksum, objectRecordStatusEntity);
+        } else {
+            log.info("Duplicate audio upload detected with no difference in file size. Returning 200 with no changes ");
+            for (MediaEntity entity : identifiedDuplicate) {
+                log.info("Duplicate media id {}", entity.getId());
             }
-        } catch (IOException e) {
-            throw new DartsApiException(FAILED_TO_UPLOAD_AUDIO_FILE, e);
         }
+    }
 
-        MediaEntity savedMedia = mediaRepository.save(mapper.mapToMedia(addAudioMetadataRequest));
-        savedMedia.setChecksum(checksum);
-        linkAudioToHearingInMetadata(addAudioMetadataRequest, savedMedia);
-        linkAudioToHearingByEvent(addAudioMetadataRequest, savedMedia);
-
-        saveExternalObjectDirectory(
-            externalLocation,
-            checksum,
-            userIdentity.getUserAccount(),
-            savedMedia,
-            objectRecordStatusEntity
-        );
+    private void versionUpload(Collection<MediaEntity> audioToVersion,
+                                AddAudioMetadataRequest addAudioMetadataRequest,
+                                UUID externalLocation, String checksum,
+                                ObjectRecordStatusEntity objectRecordStatusEntity) {
+        for (MediaEntity entity : audioToVersion) {
+
+            MediaEntity saveEntity = entity;
+
+            // if the media already exists in the database then create a new media file
+            if (entity.getId() != null) {
+                saveEntity = mapper.mapToMedia(addAudioMetadataRequest);
+
+                saveEntity.setChronicleId(entity.getChronicleId());
+                saveEntity.setAntecedentId(entity.getId().toString());
+
+                log.info("Uploading version of duplicate filename {} under version number {}", entity.getMediaFile(), entity.getLegacyVersionLabel());
+            } else {
+                log.info("New file uploaded {} with version number {}", entity.getMediaFile(), entity.getLegacyVersionLabel());
+
+                saveEntity = mediaRepository.save(saveEntity);
```

Comment:
Is this save needed if it saves again on line 213?

Reply:
for new media file uploads we set the chronicle id to the same id of the new media record. We first save to get the id and then update the chronicle id with that same id. This only happens on a new media record